packages/manager/apps/web/client/app/hosting/multisite/hosting-multisite.controller.js

```diff
@@ -105,72 +107,74 @@ angular
         $scope.existingWebsites = existingWebsites;
       });
 
-      $scope.getNichandleAttachedWithFallback = function getNichandleAttachedWithFallback(
-        domainName,
-      ) {
-        return $scope.hosting.serviceName !== domainName
-          ? HostingDomain.getNichandleAttached(
-              domainName,
-              $scope.user.nichandle,
-            )
-          : $q.resolve({ isDnsAttachedToNic: true });
+      $scope.wrapperZoneInfo = function wrapperZoneInfo(domainName) {
+        const rootDomain = domainName
+          .split('.')
+          .slice(-2)
+          .join('.');
+
+        return {
+          isOVHZone: $scope.listZone.includes(rootDomain),
+          zone: rootDomain,
+        };
       };
 
-      $scope.getZoneLinkedWithFallback = function getZoneLinkedWithFallback(
-        domainName,
-      ) {
-        return $scope.hosting.serviceName !== domainName
-          ? HostingDomain.getZoneLinked(domainName)
-          : $q.resolve(domainName);
+      $scope.wrapperDigStatus = function wrapperDigStatus({
+        records,
+        recommendedIps,
+      }) {
+        return Object.entries(records).reduce(
+          (acc, [key, { type }]) => ({
+            ...acc,
+            [type]: {
+              ip: key,
+              status: recommendedIps[
+                `recommended${RECORD_TYPE_TO_IP_TYPE[type]}`
+              ].includes(key),
+              recommendedIps:
+                recommendedIps[`recommended${RECORD_TYPE_TO_IP_TYPE[type]}`],
+            },
+          }),
+          {},
+        );
       };
 
-      $scope.storeDiagnosticInfos = function storeDiagnosticInfos(
-        domain,
-        zoneLinked,
-        record,
+      $scope.getDiagnosticBadge = function getDiagnosticBadge(
+        { digStatus },
         recordType,
-        nicProperties,
       ) {
-        $scope.diagnosticSuffix[$scope.getKeyFor(domain.name, recordType)] =
-          DIAGNOSTIC_STATE.INFO;
-        $scope.optionsForModal[$scope.getKeyFor(domain.name, recordType)] = {
-          ...$scope.optionsForModal[$scope.getKeyFor(domain.name, recordType)],
-          zone: zoneLinked.zone,
-          diagnosticState:
-            $scope.diagnosticSuffix[$scope.getKeyFor(domain.name, recordType)],
-          isDnsAttachedToNic: nicProperties.isDnsAttachedToNic,
-        };
-        $scope.optionsForModal[
-          $scope.getKeyFor(domain.name, recordType)
-        ].ip = record;
+        return DIAGNOSTIC_BADGE_STATE[recordType][
+          digStatus[recordType]?.status.toString()
```

Comment:
`toString` is provided by the API ? Why do you need it ?

Reply:
I manage the translation with _true,  _false or _undefined and i need also for the object property. because if not false and undefined it's same if i don't convert toString.